index.ts

```diff
@@ -0,0 +1,345 @@
+class ElementCreator {
+    createElement(tagName: string, attributes?: Record<string, string>): HTMLElement {
+        const element = document.createElement(tagName);
+
+        if (attributes) {
+            Object.keys(attributes).forEach((key) => {
+                element.setAttribute(key, attributes[key]);
+            });
+        }
+
+        return element;
+    }
+}
+
+class FormElementCreator {
+    private elementCreator: ElementCreator;
+
+    constructor(elementCreator: ElementCreator) {
+        this.elementCreator = elementCreator;
+    }
+
+    createInput(form: HTMLFormElement, element: any): void {
+        const input = this.elementCreator.createElement('input') as HTMLInputElement;
+        this.setInputAttributes(input, element);
+        form.appendChild(input);
+    }
+
+    createLabel(form: HTMLFormElement, element: any): void {
+        const label = this.elementCreator.createElement('label') as HTMLLabelElement;
+        label.textContent = element.for;
+        this.createInput(form, element);
+        form.appendChild(label);
+    }
+
+    createInputFields(form: HTMLFormElement, element: any): void {
+        const paragraph = this.elementCreator.createElement('div') as HTMLDivElement;
+        paragraph.textContent = element.name;
+        form.appendChild(paragraph);
+        this.createInput(form, element);
+    }
+
+    private setInputAttributes(input: HTMLInputElement, element: any): void {
+        input.type = element.type;
+        input.name = element.name;
+        element.class === 'buttons' ? (input.value = element.value) : (input.placeholder = element.placeholder);
+    }
+
+    createFormElement(form: HTMLFormElement, element: any): void {
+        switch (element.class) {
+            case 'selects':
+                this.createLabel(form, element);
+                break;
+            case 'buttons':
+                this.createInput(form, element);
+                break;
+            case 'fields':
+                this.createInputFields(form, element);
+                break;
+            default:
+                console.log('field type not recognized');
```

Comment:
tu mozesz dorzucic po przecinku jaki to byl typ podany przez uzytkownika zeby wiedzial co bylo nie tak

Reply:
`console.log('blabla', element.class)`